lib/core/connector_settings.rb

```diff
@@ -70,26 +66,26 @@ def scheduling_settings
     end
 
     def request_pipeline
-      return_if_present(dig(:pipeline, :name), @globals.dig(:pipeline, :default_name), DEFAULT_REQUEST_PIPELINE)
+      return_if_present(@elasticsearch_response.dig(:pipeline, :name), @globals.dig(:pipeline, :default_name), DEFAULT_REQUEST_PIPELINE)
     end
 
     def extract_binary_content?
-      return_if_present(dig(:pipeline, :extract_binary_content), @globals.dig(:pipeline, :default_extract_binary_content), DEFAULT_EXTRACT_BINARY_CONTENT)
+      return_if_present(@elasticsearch_response.dig(:pipeline, :extract_binary_content), @globals.dig(:pipeline, :default_extract_binary_content), DEFAULT_EXTRACT_BINARY_CONTENT)
     end
 
     def reduce_whitespace?
-      return_if_present(dig(:pipeline, :reduce_whitespace), @globals.dig(:pipeline, :default_reduce_whitespace), DEFAULT_REDUCE_WHITESPACE)
+      return_if_present(@elasticsearch_response.dig(:pipeline, :reduce_whitespace), @globals.dig(:pipeline, :default_reduce_whitespace), DEFAULT_REDUCE_WHITESPACE)
     end
 
     def run_ml_inference?
-      return_if_present(dig(:pipeline, :run_ml_inference), @globals.dig(:pipeline, :default_run_ml_inference), DEFAULT_RUN_ML_INFERENCE)
+      return_if_present(@elasticsearch_response.dig(:pipeline, :run_ml_inference), @globals.dig(:pipeline, :default_run_ml_inference), DEFAULT_RUN_ML_INFERENCE)
     end
 
     private
 
-    def initialize(es_response)
-      @elasticsearch_response = es_response.with_indifferent_access
-      @globals = ElasticConnectorActions.connectors_meta
+    def initialize(es_response, globals)
```

Comment:
I think we can have `globals` as optional:
```suggestion
    def initialize(es_response, globals=nil)
```

and have a method `globals` in `Core::ConnectorSettings`:

```
def globals
  @globals ||= ElasticConnectorActions.connectors_meta
end
```

Reply:
From what I understood from @seanstory's comment [here](https://github.com/elastic/connectors-ruby/pull/283#issuecomment-1239520285), global settings will always be present.

The calls to `connectors_meta` will be only in 2 methods that fetch connector settings: `ConnectorSettings.fetch_by_id` and `ConnectorSettings.fetch_native`(this one is now in ElasticConnectorActions), I think it should be okay to keep the calls inlined there.